common/cmake/devices.cmake

```diff
@@ -975,6 +975,11 @@ function(ADD_FPGA_TARGET)
     if(NOT "${ADD_FPGA_TARGET_INPUT_IO_FILE}" STREQUAL "")
       add_file_target(FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
     endif()
+    if(NOT "${ADD_FPGA_TARGET_INPUT_XDC_FILE}" STREQUAL "")
+      add_file_target(FILE ${ADD_FPGA_TARGET_INPUT_XDC_FILE})
+      get_file_location(INPUT_XDC_FILE ${ADD_FPGA_TARGET_INPUT_XDC_FILE})
+      get_target_property_required(PART_JSON ${BOARD} PART_JSON)
```

Comment:
`PART_JSON` will not be not be defined the option `EXPLICIT_ADD_FILE_TARGET` is specified

Reply:
https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1181#discussion_r352749731

Reply:
I changed from `get_target_property` to `get_target_property_required`. What else is needed?

Reply:
If the option `EXPLICIT_ADD_FILE_TARGET`, this part of the control flow will never be executed, and the variable `PART_JSON` will not get defined.

Reply:
This comment is the last outstanding issue

Reply:
Not sure what `EXPLICIT_ADD_FILE_TARGET` is used for?

Reply:
It is used whenever files are shared between targets.  The implicit calls to `add_file_target` result in errors if two instances of `add_fpga_target` both share the same verilog file.  For example, if the min and full LiteX share verilog files, then `EXPLICIT_ADD_FILE_TARGET` should be specified, etc.

Reply:
Is simply moving the `get_target_property_required(PART_JSON..` from inside of the `EXPLICIT_ADD_FILE_TARGET` check a solution here?

Reply:
Yes, but I only want `get_target_property_required(PART_JSON..` called if an `XDC` file is specified.

Reply:
Ok - [this](https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1158/commits/8d4a4a7cfa92c78fb64b5311b58d962ba27f00b6) should be fine then.